main.tf

```diff
@@ -0,0 +1,45 @@
+terraform {
+  required_version = ">= 0.11.11"
```

Comment:
Why is this the pinned version? You're building it with TF 0.12.2, but requiring a 0.11.x version. Seems like this should be lower, or higher.

Reply:
I'm only validating. I want this module to be compatible with Terraform 0.12.x and 0.11.x, so that I do not have to make any changes when we upgrade Terraform in other places. I think the best solution would be to validate both 0.11.x and 0.12.x. I'll do that unless you have objections.

Reply:
ping @mbrochstein How do you feel about my approach?

Reply:
Pinning the versions makes sense, but the module can't be compatible with both tf 0.11.x and tf 0.12.x because of the changes to HCL, right?

Reply:
From what I can tell, this module is compatible with both. I test on 0.12.2 and the module has already been used on 0.11.x through other repositories that pull it in. I made an effort uses of 0.12.x-only features / syntax.

Reply:
To extend upon this, I do not want to pin a module to a specific terraform version. This module is public, so I have designed it under the assumption I cannot  (well, should not) control the exact version the module requires. I would like to avoid upper-bound version constraints until it's demonstrated that there is an incompatibility with some version.